File: Angular/src/app/services/extract.service.ts
import { ExtractRepository } from '../repositories/extract.repository';
import { Injectable } from '@angular/core';
import { ExtractModel } from '../models/extract/extract.model';
import { ExtractMapper } from '../mapper/extract.mapper';

@Injectable()
export class ExtractService {
  constructor(
    private _extractRepository: ExtractRepository,
    private _extractMapper: ExtractMapper
  ) { }

  public async getDataExtract(): Promise<ExtractModel[]> {

    const view = await this._extractRepository.getDataExtract();

    const model = this._extractMapper.modelsToViewsModel(view.data);

    return model;
  }
}

File: Spring/src/main/java/com/test/service/ExtractService.java
package com.test.service;

import java.util.List;

import com.test.mapper.ExtractMapper;
import com.test.model.BaseReturn;
import com.test.model.ExtractModel.Extract;
import com.test.viewModel.ExtractViewModel;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.test.repository.ExtractRepository;

@Service
public class ExtractService {

    @Autowired
    private ExtractRepository _repository;
    @Autowired
    private ExtractMapper _mapper;

    public ExtractService(ExtractRepository repository, ExtractMapper mapper) {
        this._repository = repository;
        this._mapper = mapper;
    }

    public BaseReturn<List<ExtractViewModel>> List() {
        BaseReturn<List<ExtractViewModel>> response = new BaseReturn<List<ExtractViewModel>>();

        try {
            Extract model = _repository.List();

            List<ExtractViewModel> viewModels = _mapper.ExtractViewModels(model);

            response.Data = viewModels;
        } catch (Exception ex) {
            response.Success = false;
            response.Description = ex.getMessage();
        }

        return response;
    }
}